app/src/main/java/com/github/orkest/ui/feed/FeedActivity.kt

```diff
@@ -70,11 +63,52 @@ fun FeedActivity(viewModel: PostViewModel) {
     val coroutineScope = rememberCoroutineScope()
     val lazyListState = rememberLazyListState()
 
+
     // Remember the swipe refresh state
     val isRefreshing = remember { mutableStateOf(false) }
     val swipeRefreshState = rememberSwipeRefreshState(isRefreshing = isRefreshing.value)
 
-    refreshData(viewModel, listPosts)
+    if(FireStoreDatabaseAPI.isOnline(context)) {
+        //if there is internet connexion, get posts from the firestore database and update the cache
+        refreshData(viewModel, listPosts)
+
+        CompletableFuture.runAsync{
+            val postEntities = listPosts.value.map { post ->
+                AppEntities.Companion.PostEntity(
+                    id = (0..100000).random(),
+                    username = post.username,
+                    date = post.date,
+                    profilePicId = post.profilePicId,
+                    postDescription = post.postDescription,
+                    song = post.song,
+                    nbLikes = post.nbLikes,
+                    likeList = post.likeList,
+                    nbComments = post.nbComments,
+                    media = post.media,
+                    isMediaVideo = post.isMediaVideo
+                )
+            }
+
+            database.postsDao().insertPosts(postEntities)
+        }
+    } else {
+        //if no internet connection, get posts from the cache
+        val cachedPosts: List<AppEntities.Companion.PostEntity> = database.postsDao().getAllPosts()
+        listPosts.value = cachedPosts.map { postEntity ->
+            Post(
+                username = postEntity.username,
+                date = postEntity.date,
+                profilePicId = postEntity.profilePicId,
+                postDescription = postEntity.postDescription,
+                song = postEntity.song,
+                nbLikes = postEntity.nbLikes,
+                likeList = postEntity.likeList,
+                nbComments = postEntity.nbComments,
+                media = postEntity.media,
+                isMediaVideo = postEntity.isMediaVideo
+            )
+        }
+    }
```

Comment:
This could maybe lead to collisions : id = (0..100000).random()
Maybe autogenerate = true with @PrimaryKey